### PR TITLE
loginPage object added and login.js refactored

### PR DESCRIPTION
**Pull request by GintautasB**

### CODE REVIEW

File: cypress/integration/login.js [hunk 1]
```diff
@@ -29,31 +31,31 @@ describe('Login functionality', function() {
...
-            cy.get('.user-info__title').contains(user_login)
-            cy.get('.main-nav').find('li').should('have.length', role.navItems)
-            cy.get('[aria-labelledby="timeLoggingId"]').should('have.css', 'color', 'rgb(64, 76, 237)')
+            loginPage.visit();
+            loginPage.getLoginFormUserIdInput().click({force:true})
+            loginPage.getUserName(userLogin).click()
+            loginPage.getLoginFormUserRoleInput().click({force:true})
+            loginPage.getUserRole(role.label).click()
+            loginPage.getFormSubmitBtn().click()
+            loginPage.getUrl().should('include', '/time-logging')
```
Comment (MonkeviciuteGerda) [inline cypress/integration/login.js]: Is this still login page? I think it should be timeLogging page (from this line to down below)

File: cypress/integration/login.js [hunk 2]
```diff
@@ -29,31 +31,31 @@ describe('Login functionality', function() {
...
+            loginPage.getTitle().contains('Timesheets')
+            loginPage.getCalendar().should('be.visible')
+            loginPage.getCurrentDay().should('be.visible')
+            loginPage.getCurrentDay().contains(day)
+            loginPage.getTileForm().should('be.visible')
+            loginPage.getUserInfoTitle().contains(userLogin)
+            loginPage.getMainNav().find('li').should('have.length', role.navItems)
```
Comment (MonkeviciuteGerda) [inline cypress/integration/login.js]: What is this .find('li')? This counts as hardcoded selector. You need to move this or better attach to the getMainNav function

File: cypress/objects/loginPageObject.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+class LoginPage {
+
+    visit() { cy.visit('/') };
+    getUserValidationIndicator = () => cy.get('.Select.not-valid');
```
Comment (MonkeviciuteGerda) [inline cypress/objects/loginPageObject.js:4]: You do not return this. Does it still work?

File: cypress/objects/loginPageObject.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    getLoginButton = () => cy.get('[type="submit"]');
+    getLoginFormUserIdInput = () => cy.get('[id="loginForm.userId"]');
+    getUserName = userName => cy.get(`[aria-label="${userName}"]`);
+    getLoginFormUserRoleInput = () => cy.get('[id="loginForm.role"]');
+    getUserRole = userRole => cy.get(`[aria-label="${userRole}"]`);
+    getFormSubmitBtn = () => cy.get('[type="submit"]');
+    getUrl = () => cy.url();
```
Comment (MonkeviciuteGerda) [inline cypress/objects/loginPageObject.js]: Is this still login page?

**Review by MonkeviciuteGerda — CHANGES_REQUESTED** [review]

File: cypress/objects/loginPageObject.js (hunk 3 above)
Comment (GintautasB) [reply cypress/objects/loginPageObject.js:4]: arrow type function - always returns a value Don't need to return it again.

File: cypress/integration/login.js (hunk 2 above)
Comment (GintautasB) [reply cypress/integration/login.js]: corrected

File: cypress/integration/login.js (hunk 1 above)
Comment (GintautasB) [reply cypress/integration/login.js]: corrected. separated into 2 pages

File: cypress/objects/loginPageObject.js (hunk 4 above)
Comment (GintautasB) [reply cypress/objects/loginPageObject.js]: corrected

File: cypress/objects/loginPageObject.js (hunk 3 above)
Comment (MonkeviciuteGerda) [reply cypress/objects/loginPageObject.js:4]: Nice to know :)

**Review by MonkeviciuteGerda — APPROVED** [review]
Awesome!

**Pull request merged**